data/tests/test_nightly_signal_trap.py

```
@@ -0,0 +1,245 @@
+    script, fake = harness
+    runner = script.parent / "run_fake.sh"
+    runner.write_text(f'#!/usr/bin/env bash\nexec "{sys.executable}" "{fake}" "$1"\n')
+    runner.chmod(0o755)
+    env = {**os.environ, "FAKE_GATE_CMD": str(runner), "FAKE_GATE_MODE": mode}
+
+    # Output goes to a FILE, not a pipe. On a PID-only signal the stand-in gate
+    # is orphaned and keeps running; if it held the read end of a pipe, reading
+    # it would block until that orphan exited, which is the whole 20s it sleeps.
+    log = script.parent / f"out-{mode}-{sig}-{group}.log"
+    with log.open("w") as fh:
+        p = subprocess.Popen(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)